pkg/nsx/services/realizestate/realize_state.go

```diff
@@ -76,3 +75,33 @@ func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, inte
 		return fmt.Errorf("%s not realized", intentPath)
 	})
 }
+
+func (service *RealizeStateService) GetPolicyTier1UplinkPortIP(intentPath string) (string, error) {
+	results, err := service.NSXClient.RealizedEntitiesClient.List(intentPath, nil)
+	err = nsxutil.TransNSXApiError(err)
+	if err != nil {
+		return "", err
+	}
+
+	for _, result := range results.Results {
```

Comment:
could we filter by entity_type=RealizedLogicalRouterPort?

Reply:
Good suggestion, Thanks. 
Change code as advised.